armi/reactor/blocks.py

```diff
@@ -811,14 +811,12 @@ def completeInitialLoading(self, bolBlock=None):
         self.p.enrichmentBOL = self.getFissileMassEnrich()
         massHmBOL = 0.0
         sf = self.getSymmetryFactor()
-        for child in self.iterComponents():
-            child.p.massHmBOL = child.getHMMass() * sf  # scale to full block
-            massHmBOL += child.p.massHmBOL
-            self.p.puFrac = (
-                self.getPuMoles() / self.p.molesHmBOL
-                if self.p.molesHmBOL > 0.0
-                else 0.0
-            )
```

Comment:
> No motivation, must have gotten missed in a merge. Will re-add.

Sweet. Thanks! 

> is it intention for that update to be called for each iteration in the loop over children? That feels like an N^2 action (iterating over children, then self.getPuMoles also (I assume) iterates over children to collect Pu) for each Block

Hmmm. If I'm following you right, are you suggesting that we pull `self.p.puFrac` out of the loop over `for child in self`? If so, I think that's reasonable. We don't need to set (and reset) `self.p.puFrac` while looping over each component. 

@mgjarrett for visibility since git blame shows you were here last.

Reply:
Yep, looks wrong to me.

The only explanation I could think of is that maybe I was intending to set the `puFrac` on each child but just copy/pasted and didn't change `self` to `child`. But since the `child` doesn't have `molesHmBol` defined I think we shouldn't go that route.

It looks like we should just bump that out one level of indentation.

Reply:
@albeanth @mgjarrett thanks for the feedback. I'll re-add but one level of indentation back

Reply:
Applied in dffed2982dcdde44682633f4fafedd5050c78a76

Reply:
Major shout out to @albeanth for noting the missing `puFrac`!!